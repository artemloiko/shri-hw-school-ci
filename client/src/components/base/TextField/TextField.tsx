import React from 'react';
import { cn, CNProps } from '../../../utils/bem-cn';

import './TextField.css';

type TextFieldMods = {
  required?: boolean;
  row?: boolean;
};

type Props = {
  appendText?: string;
};

export type TextFieldProps = Props & React.HTMLProps<HTMLLabelElement> & CNProps<TextFieldMods>;

const TextField: React.FC<TextFieldProps> = (props) => {
  const { htmlFor, label, children, appendText } = props;

  return (
    <div className={cn('textfield', props)}>
      <label className="textfield__label" htmlFor={htmlFor}>
        {label}
      </label>
      {children}
      {appendText && <div className="textfield__append">{appendText}</div>}
    </div>
  );
};

export default TextField;
