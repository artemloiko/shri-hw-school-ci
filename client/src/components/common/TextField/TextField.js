import React from 'react';
import PropTypes from 'prop-types';
import { cn } from 'utils/bem-cn';

import './TextField.css';

function TextField(props) {
  const { htmlFor, label, children, appendText } = props;

  return (
    <div className={cn('textfield', props)}>
      <label className="textfield__label" htmlFor={htmlFor}>
        {label}
      </label>
      {children}
      {appendText && <div className="textfield__append">{appendText}</div>}
    </div>
  );
}

TextField.propTypes = {
  label: PropTypes.string,
  htmlFor: PropTypes.string,
  childre: PropTypes.element,

  className: PropTypes.string,
  mods: PropTypes.shape({
    required: PropTypes.bool,
    row: PropTypes.bool,
  }),
  mix: PropTypes.arrayOf(PropTypes.string),
};

export default TextField;
