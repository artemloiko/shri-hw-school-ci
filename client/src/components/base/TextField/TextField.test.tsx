import React from 'react';
import renderer from 'react-test-renderer';

import TextField from './TextField';

test('TextField with all possible props matches the snapshot', () => {
  const tree = renderer
    .create(
      <TextField
        label="Label"
        htmlFor="input"
        appendText="appendText"
        mods={{
          required: true,
          row: true,
        }}
        className="block"
        mix={['block']}
      >
        <input type="text" id="input" />
      </TextField>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
