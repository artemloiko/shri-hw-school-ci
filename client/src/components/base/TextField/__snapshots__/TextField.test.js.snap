// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField with all possible props matches the snapshot 1`] = `
<div
  className="block textfield textfield_required textfield_row block__textfield"
>
  <label
    className="textfield__label"
    htmlFor="input"
  >
    Label
  </label>
  <input
    id="input"
    type="text"
  />
  <div
    className="textfield__append"
  >
    appendText
  </div>
</div>
`;
